src/workerd/api/node/tests/http-server-nodejs-test.js

```diff
@@ -0,0 +1,573 @@
+import http from 'node:http';
+import { strictEqual, ok, throws, notStrictEqual } from 'node:assert';
+import { nodeCompatHttpServerHandler } from 'cloudflare:workers';
+import { mock } from 'node:test';
+
+export const checkPortsSetCorrectly = {
+  test(_ctrl, env) {
+    const keys = ['PONG_SERVER_PORT'];
+    for (const key of keys) {
+      strictEqual(typeof env[key], 'string');
+      ok(env[key].length > 0);
+    }
+  },
+};
+
+// Test is taken from test/parallel/test-http-server-async-dispose.js
+export const testHttpServerAsyncDispose = {
+  async test() {
+    const server = http.createServer();
+
+    server.listen(8080);
+    ok(server.listening);
+    const closeFn = mock.fn();
+    server.on('close', closeFn);
+    await server[Symbol.asyncDispose]();
+    ok(!server.listening);
+    strictEqual(closeFn.mock.callCount(), 1);
+  },
+};
+
+// Test is taken from test/parallel/test-http-server-de-chunked-trailer.js
+export const testHttpServerDeChunkedTrailer = {
+  async test(_ctrl, env) {
+    const handlerFn = mock.fn((req, res) => {
+      res.setHeader('Trailer', 'baz');
+      const trailerInvalidErr = {
+        code: 'ERR_HTTP_TRAILER_INVALID',
+        message: 'Trailers are invalid with this transfer encoding',
+        name: 'Error',
+      };
```

Comment:
This doesn't make much sense to me really. We don't support trailers at all, under any transfer encoding. The error message may be confusing/misleading for folks. I would suggest that, at least for now, any attempt to use `Trailer` or to send trailers at all should just be rejected or ignored.

Reply:
We now ignore it.